src/test/java/org/sonarsource/sonarlint/ls/standalone/PromotionalNotificationsTests.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SonarLint Language Server
+ * Copyright (C) 2009-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.ls.standalone;
+
+import java.util.List;
+import org.eclipse.lsp4j.DidOpenTextDocumentParams;
+import org.eclipse.lsp4j.TextDocumentItem;
+import org.junit.jupiter.api.Test;
+import org.sonarsource.sonarlint.ls.SonarLintExtendedLanguageClient;
+import org.sonarsource.sonarlint.ls.settings.SettingsManager;
+import org.sonarsource.sonarlint.ls.standalone.notifications.PromotionalNotifications;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class PromotionalNotificationsTests {
+  private final SonarLintExtendedLanguageClient client = mock(SonarLintExtendedLanguageClient.class);
+  private final SettingsManager settingsManager = mock(SettingsManager.class);
+  private final PromotionalNotifications underTest = new PromotionalNotifications(client, settingsManager);
+
+  @Test
+  void shouldSendNotification_notConnected_commercialLanguage() {
+    var didOpenTextDocumentParams = new DidOpenTextDocumentParams(new TextDocumentItem("file:///1/2/3.cbl", "COBOL", 1, ""));
```

Comment:
Referring to my other comment, `"COBOL"` is not the language ID sent by the client for COBOL files.

![image](https://github.com/SonarSource/sonarlint-language-server/assets/1734254/93fa86ac-39aa-4f1b-a03f-a255bc722f71)

Reply:
![Screenshot 2024-02-13 at 11 51 07](https://github.com/SonarSource/sonarlint-language-server/assets/104207951/5773a7f7-bd17-451f-b291-25173f8dc4cf)
With `COBOL` extension installed, it seems to be. In any case, comparison should not be case-sensitive